Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/StartRide.hs

```diff
@@ -121,46 +121,50 @@ startRide ::
   StartRideReq ->
   m APISuccess.APISuccess
 startRide ServiceHandle {..} rideId req = withLogTag ("rideId-" <> rideId.getId) $ do
-  ride <- findRideById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
-  let driverId = ride.driverId
-  let driverKey = makeStartRideIdKey driverId
-  Redis.setExp driverKey ride.id 60
-  rateLimitStartRide driverId ride.id -- do we need it for dashboard?
-  booking <- findBookingById ride.bookingId >>= fromMaybeM (BookingNotFound ride.bookingId.getId)
-  driverInfo <- QDI.findById (cast driverId) >>= fromMaybeM (PersonNotFound driverId.getId)
-  unless (driverInfo.subscribed) $ throwError DriverUnsubscribed
-  case req of
-    DriverReq driverReq -> do
-      let requestor = driverReq.requestor
-      case requestor.role of
-        DP.DRIVER -> unless (requestor.id == driverId) $ throwError NotAnExecutor
-        _ -> throwError AccessDenied
-    DashboardReq dashboardReq -> do
-      unless (booking.providerId == dashboardReq.merchantId) $ throwError (RideDoesNotExist ride.id.getId)
-
-  unless (isValidRideStatus (ride.status)) $ throwError $ RideInvalidStatus "This ride cannot be started"
-
-  point <- case req of
-    DriverReq driverReq -> do
-      when (driverReq.rideOtp /= ride.otp) $ throwError IncorrectOTP
-      logTagInfo "driver -> startRide : " ("DriverId " <> getId driverId <> ", RideId " <> getId ride.id)
-      pure driverReq.point
-    DashboardReq dashboardReq -> do
-      logTagInfo "dashboard -> startRide : " ("DriverId " <> getId driverId <> ", RideId " <> getId ride.id)
-      case dashboardReq.point of
-        Just point -> pure point
-        Nothing -> do
-          driverLocation <- findLocationByDriverId driverId >>= fromMaybeM LocationNotFound
-          pure $ getCoordinates driverLocation
-
-  whenWithLocationUpdatesLock driverId $ do
-    withTimeAPI "startRide" "startRideAndUpdateLocation" $ startRideAndUpdateLocation driverId ride booking.id point booking.providerId
-    withTimeAPI "startRide" "initializeDistanceCalculation" $ initializeDistanceCalculation ride.id driverId point
-    withTimeAPI "startRide" "notifyBAPRideStarted" $ notifyBAPRideStarted booking ride
-  CQDGR.setDriverGoHomeIsOnRide driverId booking.providerId
+  Redis.whenWithLockRedis (makestartRideKeyWithRideId rideId) 60 $ do
```

Comment:
Let's add it in `otpRideCreate` instead so that the ride entity itself is not created